### PR TITLE
[WIP] Handle `_routes.generated.json` file generation

### PR DESCRIPTION
**Pull request by CarmenPopoviciu**

### CODE REVIEW

File: packages/wrangler/src/pages/functions/routes-transformation.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+describe("route-paths-to-glob-patterns", () => {
+  describe("convertRoutePathsToGlobPatterns()", () => {
+    it("should pass through routes with no wildcards", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/foo"])).toEqual(["/api/foo"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo", "/api/bar"])).toEqual(["/api/foo", "/api/bar"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo", "/api/bar/foo", "/foo/bar"])).toEqual(["/api/foo", "/api/bar/foo", "/foo/bar"]);
+    });
+
+    it("should escalate a single param route to a wildcard", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/:foo"])).toEqual(["/api/*"]);
```
Comment (jahands) [inline packages/wrangler/src/pages/functions/routes-transformation.test.ts]: Need to handle routes with middleware (see internal chat)

File: packages/wrangler/src/pages/functions/routes-transformation.test.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      expect(convertRoutePathsToGlobPatterns(["/bar/:barId/foo"])).toEqual(["/bar/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/bar/:barId/foo/:fooId"])).toEqual(["/bar/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/:foo", "/bar/:barName/profile", "/foo/bar/:barId/:fooId"])).toEqual(["/api/*", "/bar/*", "/foo/bar/*"]);
+    });
+
+    it("should pass through a single wildcard route", () => {
+      expect(convertRoutePathsToGlobPatterns(["/api/:baz*"])).toEqual(["/api/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/foo/bar/:baz*"])).toEqual(["/api/foo/bar/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/api/:foo/:bar*"])).toEqual(["/api/*"]);
+      expect(convertRoutePathsToGlobPatterns(["/foo/:foo*/bar/:bar*"])).toEqual(["/foo/*"]);
```
Comment (jahands) [inline packages/wrangler/src/pages/functions/routes-transformation.test.ts]: This pattern *shouldn't* happen in the real world, but I'm glad we're testing this!

**Review by jahands — CHANGES_REQUESTED** [review]
This looks great! A couple small changes:
- Run `npm run fix` from the repo root to fix lint errors
- Handle routes with middlewares (need to append /* to the route if they don't already end with that)
- Let's change this PR to `cloudflare/wrangler2` instead of my fork

File: packages/wrangler/src/pages/functions/routes-transformation.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/**
+ * TODO can we do better naming?
+ */
+export function convertRoutePathsToGlobPatterns(routes: string[]): string[] {
```
Comment (jahands) [inline packages/wrangler/src/pages/functions/routes-transformation.ts:4]: We should probably input/output `UrlPath` here to be consistent with other routing code. Although these routes should never be derived from file paths, it doesn't hurt to be sure.

File: packages/wrangler/src/pages/functions/routes-transformation.test.ts (hunk 2 above)
Comment (CarmenPopoviciu) [reply packages/wrangler/src/pages/functions/routes-transformation.test.ts]: yeah, I wouldn't it expect IRL either, but I figured better be safe than sorry

**Pull request closed without merging**